consensus/scp/tests/mock_network/mod.rs

```diff
@@ -244,80 +216,17 @@ impl SCPNetwork {
             .lock()
             .expect("lock failed on nodes_map in broadcast");
 
-        log::trace!(
-            logger,
-            "(broadcast) node {:3} slot {:3} : {:?}",
-            msg.sender_id,
-            msg.slot_index,
-            msg.topic
-        );
+        log::trace!(logger, "(broadcast) {}", msg.to_display(),);
 
         let amsg = Arc::new(msg);
 
-        for responder_id in peers {
+        for peer_id in peers {
             nodes_map
-                .get_mut(&responder_id)
+                .get_mut(&peer_id)
                 .expect("failed to get peer from nodes_map")
                 .send_msg(amsg.clone());
         }
     }
-
-    /// Wait for this node's ledger to grow to a specific size
-    pub fn wait_for_total_values(
-        &self,
-        node_id: &NodeID,
-        values_to_collect: usize,
-        max_wait: Duration,
-    ) {
-        let mut deadline = Instant::now() + max_wait;
-        let mut prev_num_values = 0;
-        let mut last_log = Instant::now();
-
-        while Instant::now() < deadline {
-            let cur_num_values = self.get_shared_data(node_id).total_values();
-            if cur_num_values >= values_to_collect {
-                log::trace!(
-                    self.logger,
-                    "( testing ) node {:3}          : {:5} values ... WAIT COMPLETE!",
-                    node_id,
-                    values_to_collect,
-                );
-                return;
-            }
-
-            // As long as we keep seeing new values get added, reset our timeout.
-            if prev_num_values != cur_num_values {
-                assert!(cur_num_values > prev_num_values);
-                prev_num_values = cur_num_values;
-                deadline = Instant::now() + max_wait;
-            }
-
-            thread::sleep(Duration::from_millis(10));
-
-            if last_log.elapsed().as_secs() > 1 {
-                log::info!(
-                    self.logger,
-                    "Got {}/{} values from Node {}",
-                    cur_num_values,
-                    values_to_collect,
-                    node_id
-                );
-                last_log = Instant::now();
-            }
-        }
-
-        let cur_num_values = self.get_shared_data(node_id).total_values();
-        panic!(
-            "{:?}: timeout while waiting for {} total values (currently at {})",
-            node_id, values_to_collect, cur_num_values
-        );
-    }
-}
-
-impl Drop for SCPNetwork {
```

Comment:
This drop shouldn't be removed. We have discussed this many months ago and in the previous PR. If there is a bug that prevents it from working, it should be addressed but simply removing this is not an acceptable workaround in my opinion.